selfservice/flow/recovery/flow.go

```diff
@@ -167,6 +180,10 @@ func (f Flow) GetNID() uuid.UUID {
 	return f.NID
 }
 
+func (f Flow) ShouldEnforceCSRF() bool {
+	return f.Type.IsBrowser() && f.CSRFToken != ""
```

Comment:
The original assertion

```
	if f.Type == flow.TypeBrowser && !nosurf.VerifyToken(h.d.GenerateCSRFToken(r), f.CSRFToken) {
```

was weakened quite a bit, because now an empty `f.CSRFToken` disables CSRF validation. But is that really what this state expresses? Disabling security features should always be explicit - default values (empty string) should not allow insecure behaviour! I suggest to revert this and instead make sure that callers decide when it is ok to not have an CSRF Token

Reply:
The reasoning here is, that we control the creation of the `flow` and thus if a CSRF token is required. And since the new flow now always requires CSRF protection at all steps, _except_ when the flow was created by an admin, it would be fair to assume, that an empty `CSRFToken` means that a flow was created by an admin. So: yes, this is what that state expresses.

I'd say a more explicit solution could be, to add a `CreatedBy` = `ADMIN` | `SELFSERVICE` field to `Flow` and then enforce CSRF on the `SELFSERVICE` flows. Thoughts?

Reply:
Also, another thought: if there is no `CSRFToken` in the database, how would we validate it anyway? Would it be fine to assume that the flow is FUBAR at that point?

Reply:
Done https://github.com/ory/kratos/pull/2645/commits/d865fa68a978ae7dc6a8127f7e01f6145b22c390